### PR TITLE
Add code infrastructure to validate file content

### PR DESCRIPTION
**Pull request by Migaroez**
### Testing
Unit tests have been added, but for end to end testing you can add a class like the one below

```
namespace Umbraco.Cms.Core.Security;

public class SvgXssSecurityAnalyzer : IFileStreamSecurityAnalyzer
{

    public bool FileContentMatchesFileType(Stream fileStream)
    {
        var startBuffer = new byte[256];
        var endBuffer = new byte[256];
        fileStream.Read(startBuffer);
        if (endBuffer.Length > fileStream.Length)
            fileStream.Seek(0, SeekOrigin.Begin);
        else
            fileStream.Seek(fileStream.Length - endBuffer.Length, SeekOrigin.Begin);
        fileStream.Read(endBuffer);
        var startString = System.Text.Encoding.UTF8.GetString(startBuffer);
        var endString = System.Text.Encoding.UTF8.GetString(endBuffer);
        return startString.Contains("<svg")
               && startString.Contains("xmlns=\"http://www.w3.org/2000/svg\"")
               && endString.Contains("/svg>");
    }

    /// <summary>
    /// Analyzes whether the svg contains any XSS vulnerability
    /// </summary>
    /// <param name="fileStream">Needs to be a Read/Write seekable stream</param>
    /// <returns>Whether the file is considered safe</returns>
    public bool IsConsideredSafe(Stream fileStream)
    {
        var streamReader = new StreamReader(fileStream); // do not use a using as this will dispose of the underlying stream
        var fileContent = streamReader.ReadToEnd();
        return !(fileContent.Contains("<script") && fileContent.Contains("/script>"));
    }
}
```

register it
`Services.AddSingleton<IFileStreamSecurityAnalyzer, SvgXssSecurityAnalyzer>();`

and then upload an item to the media (in any way you see fit) or user avatar with the following content
```
<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">

<svg version="1.1" baseProfile="full" xmlns="http://www.w3.org/2000/svg">
   <polygon id="triangle" points="0,0 0,50 50,0" fill="#009900" stroke="#004400"/>
   <script type="text/javascript">
      alert('xss');
   </script>
</svg>
```

### CODE REVIEW

**Comment by Migaroez** [comment]
Added docs
https://github.com/umbraco/UmbracoDocs/pull/5344

File: src/Umbraco.Core/Security/FileStreamSecurityValidator.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        {
+            fileStream.Seek(0, SeekOrigin.Begin);
+            if (!fileSanitizer.FileContentMatchesFileType(fileStream))
+            {
+                continue;
+            }
+
+            fileStream.Seek(0, SeekOrigin.Begin);
+            if (fileSanitizer.IsConsideredSafe(fileStream) == false)
+                return false;
```
Comment (nikolajlauridsen) [inline src/Umbraco.Core/Security/FileStreamSecurityValidator.cs:29]:
```suggestion
            if (fileSanitizer.IsConsideredSafe(fileStream) == false)
            {
                return false;
            }
```

A bit of a nitpicky thing, but we always use brackets 😄

File: src/Umbraco.Infrastructure/PropertyEditors/FileUploadPropertyValueEditor.cs [hunk 2]
```diff
@@ -147,6 +151,9 @@ internal class FileUploadPropertyValueEditor : DataValueEditor
 
         using (FileStream filestream = File.OpenRead(file.TempFilePath))
         {
+            if (_fileStreamSecurityValidator.IsConsideredSafe(filestream) == false)
+                return null;
```
Comment (nikolajlauridsen) [inline src/Umbraco.Infrastructure/PropertyEditors/FileUploadPropertyValueEditor.cs:155]:
```suggestion
            if (_fileStreamSecurityValidator.IsConsideredSafe(filestream) == false)
            {
                return null;
            }
```

File: src/Umbraco.Core/Security/IFileStreamSecurityAnalyzer.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+namespace Umbraco.Cms.Core.Security;
+
+public interface IFileStreamSecurityAnalyzer
+{
+
+    bool FileContentMatchesFileType(Stream fileStream);
```
Comment (nikolajlauridsen) [inline src/Umbraco.Core/Security/IFileStreamSecurityAnalyzer.cs:6]: Maybe we should rename this to `ShouldHandle`

File: src/Umbraco.Infrastructure/PropertyEditors/ImageCropperPropertyValueEditor.cs [hunk 4]
```diff
@@ -236,6 +240,9 @@ public override string ConvertDbToString(IPropertyType propertyType, object? val
 
         using (FileStream filestream = File.OpenRead(file.TempFilePath))
         {
+            if (_fileStreamSecurityValidator.IsConsideredSafe(filestream) == false)
+                return null;
```
Comment (nikolajlauridsen) [inline src/Umbraco.Infrastructure/PropertyEditors/ImageCropperPropertyValueEditor.cs:244]:
```suggestion
            if (_fileStreamSecurityValidator.IsConsideredSafe(filestream) == false)
            {
                return null;
            }
```

**Review by nikolajlauridsen — COMMENTED** [review]
Look beautiful, only a couple of suggestions 😄

**Review by nikolajlauridsen — APPROVED** [review]
Looks good, tests good, I'll go ahead and merge 😄

**Pull request merged**